strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/StrongboxBuiltinAuthenticationSupplier.java

```diff
@@ -0,0 +1,95 @@
+package org.carlspring.strongbox.authentication.api.impl;
+
+import org.carlspring.strongbox.authentication.api.AuthenticationSupplier;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+
+import com.google.common.base.Throwables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.crypto.codec.Base64;
+import org.springframework.util.Assert;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class StrongboxBuiltinAuthenticationSupplier
+        implements AuthenticationSupplier
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(StrongboxBuiltinAuthenticationSupplier.class);
+
+    private String credentialsCharset = "UTF-8";
+
+    @CheckForNull
+    @Override
+    public Authentication supply(@Nonnull HttpServletRequest request)
+    {
+        final String header = request.getHeader("Authorization");
+
+        if (header == null || !header.startsWith("Basic "))
+        {
+            return null;
+        }
+
+        String[] tokens;
+        try
+        {
+            tokens = extractAndDecodeHeader(header);
+        }
+        catch (UnsupportedEncodingException e)
+        {
+            throw Throwables.propagate(e);
+        }
+
+        String username = tokens[0];
+
+        logger.debug("Basic Authentication Authorization header found for user '" + username + "'");
+
+        return new UsernamePasswordAuthenticationToken(username, tokens[1]);
+    }
+
+    public void setCredentialsCharset(String credentialsCharset)
+    {
+        Assert.hasText(credentialsCharset, "credentialsCharset cannot be null or empty");
```

Comment:
Correct me, if I'm wrong, but as this is not test code, I don't think assertions are a good idea here. Wouldn't `StringUtils` have more effect instead?